woocommerce/google-listings-and-ads
# Attribute Mapping - Syncer

### Changes proposed in this Pull Request:

Closes part of #1648 .

This PR implements a Syncer logic to Update the products once the Attribute Mapping Rules change.

- Create hooks on delete/update/create rule
- Create logic for run delayed create batch for updating the products
- Create API to check if the current process is syncing in order to show that information in the frontend

### Steps for testing

1. Checkout this PR and delete/update/create a new rule in Attribute Mapping.
2. Go to WooCommerce- Status - Scheduled Actions and see how `gla/jobs/update_all_products/create_batch` was scheduled delayed in 30 minutes.
3. Request  `GET /wp-json/wc/gla/mc/mapping/sync`. Yo will receive `is_syncing: true` as response
4. Run manually the action (or wait 30 minutes)
5. Request  `GET /wp-json/wc/gla/mc/mapping/sync`. Yo will receive `is_syncing: false` as response
